### PR TITLE
UI dropdown

### PR DESCRIPTION
**Pull request by GidJHock**
Apologies for the atrocious state of this pull request, it _should_ show you everything I did. The result isn't pretty, complex, or super impressive, but it accomplishes everything we need it to for the demonstration. This is most of the implementation done, except for the methods specifically tied to the Bed, Curtain, DoorLock, and Light devices. Once you guys have a chance to publish those changes, I should be able to finish this up in time for the presentation Tuesday.

### CODE REVIEW

**Comment by nwestervelt** [comment]
FYI you have to add reviewers to your PRs, otherwise no one can review your PR. On the GitHub website, you can do that by clicking on the cog icon next to "Reviewers" at the top right of a PR's page.

I can add my review now, but I don't know if other people can review the PR when you haven't requested that they submit a review for your PR.

**Comment by nwestervelt** [comment]
> > FYI you have to add reviewers to your PRs, otherwise no one can review your PR. On the GitHub website, you can do that by clicking on the cog icon next to "Reviewers" at the top right of a PR's page.
> > I can add my review now, but I don't know if other people can review the PR when you haven't requested that they submit a review for your PR.
> 
> You know, I looked at that option this morning as I was making the pull request and thought, "Nah, that must just be some weird GitHub feature, surely that won't do what I think it will."

I think you mistakenly made everyone an assignee instead of a reviewer.

**Comment by nwestervelt** [comment]
I have a suggestion, adding an action listener to your combo box to refresh the UserInterface's data about devices. This is to make sure that when multiple UserInterface instances are simultaneously connected that they always have up to date information before listing the devices within the combo box.

There's already an action listener (named ActionHandler) class in the UserInterface, and there's already an instance of it called `ah`, so adding one to your combo box should be as simple as `nameOfComboBox.addActionListener(ah);`.

You would also have to add some code within the ActionHandler class to check if the ComboBox was the source of the event, but that shouldn't be too hard.

**Comment by nwestervelt** [comment]
If someone reviews this PR and then you push a new commit onto the branch, then their review will no longer be up to date (or work to meet the branch protection policy).

Pushing new commits onto your branch is all you need to do to add changes to a PR after it's been created.

**Comment by nwestervelt** [comment]
I could make suggestions on GitHub that you can commit to get this done, if you would prefer to do that.

File: UserInterface.java [hunk 1]
```diff
@@ -22,29 +30,66 @@ public UserInterface()
...
 
+        JPanel cbPanel = new JPanel();
+        mainPanel.add(cbPanel); 
+        
+        expandButton= new JButton("Expand");
+        expandButton.addActionListener(ah);
+        expandButton.setEnabled(false);
+
+        JLabel cbLabel = new JLabel("Devices: ");
+        cbPanel.add(cbLabel);
```
Comment (nwestervelt) [inline UserInterface.java:62]: Example of a suggestion that you can commit.
```Suggestion
        cbPanel.add(cbLabel);
        
```

**Comment by StitchsFriend** [comment]
reviewed

**Review by tprather12 — DISMISSED** [review]

**Review by Rykrenon — APPROVED** [review]

**Review by StitchsFriend — APPROVED** [review]

**Pull request merged**